framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -872,6 +896,18 @@ public AnnotationMirror leastUpperBound(AnnotationMirror a1, AnnotationMirror a2
          */
         @Override
         public boolean isSubtype(AnnotationMirror subAnno, AnnotationMirror superAnno) {
+            if (propertyFileHandler != null) {
+                if (inPropertyFileQualifierHierarchy(subAnno)
+                        && inPropertyFileQualifierHierarchy(superAnno)) {
+                    return propertyFileHandler.isSubtype(subAnno, superAnno);
+                } else if (inValueQualifierHierarchy(subAnno)
+                        && inPropertyFileQualifierHierarchy(superAnno)) {
+                    return false;
+                } else if (inPropertyFileQualifierHierarchy(subAnno)
+                        && inValueQualifierHierarchy(superAnno)) {
+                    return false;
+                }
```

Comment:
Add a comment that you fall through when both are in the value hierarchy.

Reply:
I think you mean: "Fall through when two are in different type hierarchy", right?

E.g., one is in `propfile` hierarchy, another in `value` hierarchy, then return `false`.

If two are in `value` type hierarchy, then we should continue executing the rest of the method `isSubType`.